api/src/main/java/org/openmrs/module/fhir2/providers/r4/PatientFhirResourceProvider.java

```diff
@@ -98,6 +100,16 @@ public MethodOutcome updatePatient(@IdParam IdType id, @ResourceParam Patient pa
 		return FhirProviderUtils.buildUpdate(patientService.update(id.getIdPart(), patient));
 	}
 	
+	@Patch
+	public MethodOutcome patchPatient(@IdParam IdType id, PatchTypeEnum patchType, @ResourceParam String body) {
```

Comment:
@ibacher i have tried testing this json merge patch using http://localhost:8091/openmrs/ws/fhir2/R4/Patient/f2dc257b-9333-4eb3-afb8-da63b689bfe8 but it seems to be throwing a `"'Patient#7' failed to validate with reason: uuid: This value exceeds the maximum length of 38 permitted for this field."`. 

Here is a copy of the request via postman https://www.postman.com/galactic-meadow-218414/workspace/gsoc2023-fhir/request/16647760-39aba897-6333-4e5f-9e20-e94f890a0433

/cc: @abertnamanya

Reply:
Well, that's unexpected! Usually, when I run into these sorts of issues, I setup an integration test that will run the code and then step through it with a debugger to figure out where things go wrong. The IT Mark put together [here](https://github.com/openmrs/openmrs-module-fhir2/blob/8373a4b9cb3bfa3be6b435debb38cab87c7fc444/integration-tests-2.2/src/test/java/org/openmrs/module/fhir2/provider/r4/MedicationRequestFhirResourceProvider_2_2IntegrationTest.java#L77) would be a good model for what the test could look like. You'll probably want to set a break-point [inside the `patch()` method in BaseFhirService](https://github.com/openmrs/openmrs-module-fhir2/blob/8373a4b9cb3bfa3be6b435debb38cab87c7fc444/api/src/main/java/org/openmrs/module/fhir2/api/impl/BaseFhirService.java#L130) and step through to see if you can figure out:

1. Where the UUID field is being changed to (this might be hard)
2. What the UUID field is set to at the point `applyUpdate()` is called
3. What the UUID field is set to right at the `createOrUpdate()` call [here](https://github.com/openmrs/openmrs-module-fhir2/blob/8373a4b9cb3bfa3be6b435debb38cab87c7fc444/api/src/main/java/org/openmrs/module/fhir2/api/impl/BaseFhirService.java#LL183C21-L183C21)

Reply:
Yeah, will do that. It seems to be something specific to the patient resource since I've tried it out with the location resource and the patch is successful.

Reply:
Maybe something to do with the fact that Patient extends Person or something? It shouldn't matter (because UUID is usually derived from `BaseOpenmrsObject`), but that's the one oddity about Patient that comes to mind

Reply:
@ibacher it looks to be something to do with `BaseOpenmrsObject` since it clearly gives the maximum length of the uuid to be 38 i.e 
```
@Column(name = "uuid", unique = true, nullable = false, length = 38, updatable = false)
private String uuid = UUID.randomUUID().toString();
```

However we seem to be exceeding that maximum length with the fact that instead of having `61b38324-e2fd-4feb-95b7-9e9a2a4400df` as the uuid, we instead have `Patient/61b38324-e2fd-4feb-95b7-9e9a2a4400df` with the length of 44 hence the request throwing the error mentioned [above](https://github.com/openmrs/openmrs-module-fhir2/pull/476#discussion_r1192978000).

**Details**
The issue is more to do with the fact that the [existing object](https://github.com/openmrs/openmrs-module-fhir2/blob/8373a4b9cb3bfa3be6b435debb38cab87c7fc444/api/src/main/java/org/openmrs/module/fhir2/api/impl/BaseFhirService.java#L136) has a uuid set as `61b38324-e2fd-4feb-95b7-9e9a2a4400df` and everything else is perfect at this stage. And even after going through the [translator](https://github.com/openmrs/openmrs-module-fhir2/blob/8373a4b9cb3bfa3be6b435debb38cab87c7fc444/api/src/main/java/org/openmrs/module/fhir2/api/impl/BaseFhirService.java#L142), the information in the existingfhirobject is the same which is perfect(right?).

Reply:
The [updatedFhirObject](https://github.com/openmrs/openmrs-module-fhir2/blob/8373a4b9cb3bfa3be6b435debb38cab87c7fc444/api/src/main/java/org/openmrs/module/fhir2/api/impl/BaseFhirService.java#L143) is where things seem to change because it is at this point that we have the id change to `Patient/61b38324-e2fd-4feb-95b7-9e9a2a4400df`

Checkout the screenshots below to show variations before and after translation.

Before
![Screenshot from 2023-05-17 14-33-08](https://github.com/openmrs/openmrs-module-fhir2/assets/47120265/4c9f99b2-3cde-4254-8853-337d2ea1eb2b)


After
![Screenshot from 2023-05-17 15-00-42](https://github.com/openmrs/openmrs-module-fhir2/assets/47120265/575ba9d8-4831-400a-9c8e-5e000bc1f71e)